api/src/main/java/org/openmrs/api/db/UserDAO.java

```diff
@@ -40,6 +40,16 @@
 	 */
 	public User getUserByUsername(String username) throws DAOException;
 	
+	/**
+	 * @see org.openmrs.api.UserService#getUserByEmail(java.lang.String)
+	 */
+	public User getUserByEmail(String email) throws DAOException;
```

Comment:
I know we add throws clause for DAOException historically but it's not necessary since it's a runtime exception